### PR TITLE
Managed addons check

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4267

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/validation_test.go [hunk 1]
```diff
@@ -546,10 +546,26 @@ var _ = Describe("ClusterConfig validation", func() {
...
+				It("it returns an error including which addons are missing", func() {
+					ipv6 := string(api.IPV6Family)
+					cfg.VPC.IPFamily = &ipv6
+					cfg.Addons = append(cfg.Addons,
+						&api.Addon{Name: "kube-proxy"},
+					)
+					err = cfg.ValidateVPCConfig()
+					Expect(err).To(MatchError(ContainSubstring("managed addons must be defined in case of IPv6; missing addon(s): vpc-cni, coredns")))
+				})
+			})
```
Comment (aclevername) [inline pkg/apis/eksctl.io/v1alpha5/validation_test.go:568]: nice test

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 2]
```diff
@@ -169,10 +169,34 @@ func (c *ClusterConfig) ValidateVPCConfig() error {
 		if *v != string(IPV4Family) && *v != string(IPV6Family) {
 			return fmt.Errorf("invalid value %s for ipFamily; allowed are %s and %s", *v, IPV4Family, IPV6Family)
 		}
+		// This is the new vpc check, I need this check when the user sets it.
+		if *v == string(IPV6Family) {
+			if missing := c.addonContainsManagedAddons([]string{"vpc-cni", "coredns", "kube-proxy"}); len(missing) != 0 {
+				return fmt.Errorf("managed addons must be defined in case of IPv6; missing addon(s): %s", strings.Join(missing, ", "))
```
Comment (aclevername) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:175]: "managed addons" is a bit vague, we need some other language to convey "the default core addons that must be installed"

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 3]
```diff
@@ -169,10 +169,34 @@ func (c *ClusterConfig) ValidateVPCConfig() error {
 		if *v != string(IPV4Family) && *v != string(IPV6Family) {
 			return fmt.Errorf("invalid value %s for ipFamily; allowed are %s and %s", *v, IPV4Family, IPV6Family)
 		}
+		// This is the new vpc check, I need this check when the user sets it.
+		if *v == string(IPV6Family) {
+			if missing := c.addonContainsManagedAddons([]string{"vpc-cni", "coredns", "kube-proxy"}); len(missing) != 0 {
```
Comment (nikimanoledaki) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:174]: could we use consts for the addons? 😇

Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/validation.go:174]: I was so about to add it, but noticed that nothing else is doing that but call them by name and I was lazy. :DDDDDDD But for you, I'll add it :D

File: userdocs/src/usage/vpc-networking.md [hunk 4]
```diff
@@ -40,9 +40,14 @@ metadata:
...
   ipFamily: IPv6 # or IPv4
+
+addons:
+  - name: vpc-cni
+  - name: coredns
+  - name: kube-proxy
 ```
 
-This is an in config file setting only. The default value is `IPv4`.
+This is an in config file setting only and managed addons need to be defined when IPv6 is set. The default value is `IPv4`.
```
Comment (aclevername) [inline userdocs/src/usage/vpc-networking.md:50]:
```suggestion
This is an in config file setting only and default core addons need to be defined when IPv6 is set. The default value is `IPv4`.
```

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 5]
```diff
@@ -325,6 +325,13 @@ const (
 	IPV6Family IPFamily = "IPv6"
 )
 
+// Values for core addons
+const (
+	VPCCNIAddon    = "vpc-cni"
```
[the remark this answers is not in the document]
Comment (nikimanoledaki) [reply pkg/apis/eksctl.io/v1alpha5/types.go:330]: ooo yeah that sounds bad 😬

**Pull request merged**